packages/launch_repository/lib/src/launch_repository.dart

```diff
@@ -0,0 +1,26 @@
+import 'package:spacex_api/spacex_api.dart';
+
+///Thrown when an error occurs while looking up for launches
+class LaunchException implements Exception {}
+
+/// {@template launch_repository}
+/// A Dart package to manage the launches
+/// {@endtemplate}
+class LaunchRepository {
+  /// {@macro launches_repository}
+  LaunchRepository({SpaceXApiClient? spaceXApiClient})
+      : _spaceXApiClient = spaceXApiClient ?? SpaceXApiClient();
+
+  final SpaceXApiClient _spaceXApiClient;
+
+  /// Returns the latest launch.
+  ///
+  /// Throws a [LaunchException] if an error occurs.
+  Future<Launch> fetchLatestLaunch() {
+    try {
+      return _spaceXApiClient.fetchLatestLaunch();
```

Comment:
```suggestion
      return await _spaceXApiClient.fetchLatestLaunch();
```

I believe we'd need to await the future in order to catch any exceptions.

Reply:
It only awaits the future in the presentation layer, like in the "launches_cubit". If this is true for catching any exceptions, we also have to change it, e.g. in the rocket_repository too.